src/tokens/slice.ts

```diff
@@ -129,6 +143,44 @@ const slice = createSlice({
       loading: false,
       error: false,
     }),
+    fetchPriceHistoryStart: (
+      state,
+      action: PayloadAction<{
+        tokenId: string
+        startTimestamp: number
+        endTimestamp: number
+      }>
+    ) => {
+      const { tokenId } = action.payload
+      // @ts-expect-error 'string' can't be used to index type 'WritableDraft<{}>'
```

Comment:
Not so sure about this, but since we don't have the list of tokens when initializing state I didn't know a better way. Open to any feedback or suggestions.

Reply:
you may need to make lastDay optional, but I haven't really looked into the consequences of doing that